zbus/src/connection/builder.rs

```diff
@@ -88,6 +89,22 @@ impl<'a> Builder<'a> {
         Self::address(&crate::address::system()?)
     }
 
+    /// Create a builder for an IBus connection
+    pub fn ibus() -> Result<Self> {
+        let ibus_address = {
+            let output = Command::new("ibus")
+                .arg("address")
+                .output()
+                .expect("Fail to run `ibus address`");
+            String::from_utf8(output.stdout)
```

Comment:
`std::str::from_utf8` is sufficient here and avoids an allocation/copy. You just have to break this statement into two, so you save the output somewhere first and then create a `&str` ref to that.

Reply:
I make a change. Please have a look!

I learnt Rust and long time not programming with Rust. Thanks for pointing it out!

Reply:
> I make a change. Please have a look!

Will do, thanks.



> I learnt Rust and long time not programming with Rust. Thanks for pointing it out!

No worries.

Reply:
> > I make a change. Please have a look!
> 
> Will do, thanks.

You didn't make the change to `std::str::from_utf8` and given you added another scope, you're forcing yourself to use the allocating/copying `String` API. Also see [my other comment](https://github.com/dbus2/zbus/pull/1040/files#r1788132053).